mono/mini/debugger-agent.c

```diff
@@ -4664,6 +4664,42 @@ ss_update (SingleStepReq *req, MonoJitInfo *ji, SeqPoint *sp, DebuggerTlsData *t
 	MonoDebugSourceLocation *loc = NULL;
 	gboolean hit = TRUE;
 
+	if ((ss_req->filter & STEP_FILTER_STATIC_CTOR)) {
+		mono_thread_state_init_from_monoctx (&tls->context, ctx);
```

Comment:
Wouldn't this slow down single stepping a lot, since this filter is set by the ide usually ?

Reply:
I don't believe that we have a more lightweight way to find the stack trace

Reply:
We could have an enter/exit callback that tracks how many static constructor frames (that aren't start_method) are on the stack. That might avoid having to use this machinery.